src/test/java/reactor/kafka/receiver/KafkaReceiverTest.java

```diff
@@ -192,6 +192,39 @@ public void seekToOffset() throws Exception {
         sendReceive(kafkaFlux, count, count, partitions, count * 2 - partitions);
     }
 
+    /**
+     * Consume from an offset of partitions by seeking to a timestamp for all partitions in the assign listener.
+     */
+    @Test
+    public void seekToTimestamp() throws Exception {
+        int count = 10;
+        sendMessages(0, count);
+        Thread.sleep(50);
+        long t1 = System.currentTimeMillis();
+        sendMessages(count, count);
+        receiverOptions = receiverOptions
+                .consumerProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
+                .addAssignListener(parts -> {
+                    for (ReceiverPartition p : parts) {
+                        p.seekToTimestamp(t1);
+                    }
+                    this.assignSemaphore.release();
+                })
+                .subscription(Collections.singleton(topic));
+        KafkaReceiver<Integer, String> receiver = KafkaReceiver.create(receiverOptions);
+        Flux<? extends ConsumerRecord<Integer, String>> kafkaFlux = receiver
+                .receive()
+                .doOnError(e -> log.error("KafkaFlux exception", e));
+        CountDownLatch latch = new CountDownLatch(count);
+        subscribe(kafkaFlux, latch);
+
+        waitForMessages(latch);
+        this.receivedRecords.forEach(list -> list.forEach(rec ->
+                assertThat(rec.timestamp()).isGreaterThanOrEqualTo(t1)));
+        assertThat(this.receivedRecords.stream().flatMap(list -> list.stream()).count()).isEqualTo(count);
```

Comment:
nit: add `.as("sum of list sizes")` before .isEqualTo as a way to see what the int represents in case of failure (otherwise the generic assertion error is just "expected x, got y")